commons-numbers-gamma/src/main/java/org/apache/commons/numbers/gamma/RegularizedBeta.java

```diff
@@ -0,0 +1,140 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.numbers.gamma;
+
+import org.apache.commons.numbers.fraction.ContinuedFraction;
+
+/**
+ * <a href="http://mathworld.wolfram.com/RegularizedBetaFunction.html">
+ * Regularized Beta function</a>.
+ * <p>
+ * This class is immutable.
+ * </p>
+ */
+public final class RegularizedBeta {
+    /** Maximum allowed numerical error. */
+    private static final double DEFAULT_EPSILON = 1e-14;
+
+    /** Private constructor. */
+    private RegularizedBeta() {
+        // intentionally empty.
+    }
+
+    /**
+     * Computes the value of the
+     * <a href="http://mathworld.wolfram.com/RegularizedBetaFunction.html">
+     * regularized beta function</a> I(x, a, b).
+     *
+     * @param x Value.
+     * @param a Parameter {@code a}.
+     * @param b Parameter {@code b}.
+     * @return the regularized beta function I(x, a, b).
+     * @throws ArithmeticException if the algorithm fails to converge.
+     */
+    public static double value(double x,
+                               double a,
+                               double b) {
+        return value(x, a, b, DEFAULT_EPSILON, Integer.MAX_VALUE);
+    }
+
+
+    /**
+     * Computes the value of the
+     * <a href="http://mathworld.wolfram.com/RegularizedBetaFunction.html">
+     * regularized beta function</a> I(x, a, b).
+     *
+     * The implementation of this method is based on:
+     * <ul>
+     *  <li>
+     *   <a href="http://mathworld.wolfram.com/RegularizedBetaFunction.html">
+     *   Regularized Beta Function</a>.
+     *  </li>
+     *  <li>
+     *   <a href="http://functions.wolfram.com/06.21.10.0001.01">
+     *   Regularized Beta Function</a>.
+     *  </li>
+     * </ul>
+     *
+     * @param x the value.
+     * @param a Parameter {@code a}.
+     * @param b Parameter {@code b}.
+     * @param epsilon When the absolute value of the nth item in the
+     * series is less than epsilon the approximation ceases to calculate
+     * further elements in the series.
+     * @param maxIterations Maximum number of "iterations" to complete.
+     * @return the regularized beta function I(x, a, b).
+     * @throws ArithmeticException if the algorithm fails to converge.
+     */
+    public static double value(double x,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.